backend/app/adapter/routing/handle/search.go

```diff
@@ -1,17 +1,169 @@
 package handle
 
 import (
+	"encoding/json"
+	"errors"
+	"io/ioutil"
 	"net/http"
 
 	"github.com/short-d/app/fw/router"
+	"github.com/short-d/short/backend/app/entity"
+	"github.com/short-d/short/backend/app/usecase/authenticator"
 	"github.com/short-d/short/backend/app/usecase/search"
+	"github.com/short-d/short/backend/app/usecase/search/order"
 )
 
-// Search fetches resources under certain criterias.
+// SearchRequest represents the request received from Search API.
+type SearchRequest struct {
+	query  search.Query
+	filter search.Filter
+}
+
+// Filter represents the filter field received from Search API.
+type Filter struct {
+	maxResults int
+	resources  []search.Resource
+	orders     []order.By
+}
+
+// Search fetches resources under certain criteria.
 func Search(
-	search search.Search,
+	searcher search.Search,
+	authenticator authenticator.Authenticator,
 ) router.Handle {
 	return func(w http.ResponseWriter, r *http.Request, params router.Params) {
-		w.Write([]byte("not implemented"))
+		user := getUser(r, authenticator)
+		if user == nil {
+			w.Write([]byte("user not authenticated"))
+			return
+		}
+
+		searchBody, err := parseSearchBody(r)
+		if err != nil {
+			http.Error(w, err.Error(), 500)
+			return
+		}
+		searchBody.setUser(user)
+
+		w.Write([]byte("request is read"))
+	}
+}
+
+// UnmarshalJSON implements the json.Unmarshaler interface.
+func (s *SearchRequest) UnmarshalJSON(data []byte) error {
+	temp := struct {
+		Query  string `json:"query"`
+		Filter Filter `json:"filter"`
+	}{}
+
+	if err := json.Unmarshal(data, &temp); err != nil {
+		return err
+	}
+
+	s.setQuery(temp.Query)
```

Comment:
I think both of those are parsed automatically and we don't have to explicitly `UnmarshalJSON` for `SearchRequest`. Let's put `json` tag on the request and filter themselves.